main.cpp

```diff
@@ -104,199 +114,474 @@ good to go!
+    DoubleType& add(const IntType f);
+
+    DoubleType& subtract(double d );
+    DoubleType& subtract(const FloatType f);
+    DoubleType& subtract(const IntType i);
+
+    DoubleType& multiply(double d );
+    DoubleType& multiply(const FloatType f);
+    DoubleType& multiply(const IntType i);
     
-    // Intercept division by 0
-    // --------
-    std::cout << "Intercept division by 0 " << std::endl;
-    std::cout << "New value of it = it / 0 = " << it.divide(0).value << std::endl;
-    std::cout << "New value of ft = ft / 0 = " << ft.divide(0).value << std::endl;
-    std::cout << "New value of dt = dt / 0 = " << dt.divide(0).value << std::endl;
+    DoubleType& divide(double d );
+    DoubleType& divide(const FloatType f);
+    DoubleType& divide(const IntType i);
+};
 
-    std::cout << "---------------------\n" << std::endl; 
 
-    std::cout << "good to go!\n";
+//-------------------------------------------------
+// IntType
+//-------------------------------------------------
 
-    return 0;
+struct IntType
+{
+    int* value;
+
+    IntType(int i) : value(new int{i}) {}
+    ~IntType()
+    {
+        //delete value;
+        value = nullptr;
+    }
+
+    IntType& add(int i );
+    IntType& add(const FloatType f );
+    IntType& add(const DoubleType d );
+
+    IntType& subtract(int i );
+    IntType& subtract(const FloatType f );
+    IntType& subtract(const DoubleType d );
+
+    IntType& multiply(int i );
+    IntType& multiply(const FloatType f );
+    IntType& multiply(const DoubleType d );
+
+    IntType& divide(int i );
+    IntType& divide(const FloatType f );
+    IntType& divide(const DoubleType d );
+
+};
+
+
+
+//-------------------------------------------------
+// add
+//-------------------------------------------------
+
+FloatType& FloatType::add(float rhs )
+{
+    *value += rhs;
+    return *this;
 }
 
+FloatType& FloatType::add(const DoubleType d )
+{
+    *value += *d.value;
+    return *this;
+}
 
+FloatType& FloatType::add(const IntType i)
+{
+    *value += *i.value;
+    return *this;
+}
 
-/*
- MAKE SURE YOU ARE NOT ON THE MASTER BRANCH
+//-------------------------------------------------
 
- Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
- 
- If you didn't already: 
-    Make a pull request after you make your first commit
-    pin the pull request link and this repl.it link to our DM thread in a single message.
+DoubleType& DoubleType::add(double d )
+{
+    *value += d;
+    return *this;
+}
 
- send me a DM to review your pull request when the project is ready for review.
+DoubleType& DoubleType::add(const FloatType ft)
+{
+    *value += *ft.value;
+    return *this;
+}
 
- Wait for my code review.
- */
+DoubleType& DoubleType::add(const IntType it)
+{
+    *value += *it.value;
+    return *this;
+}
+
+//-------------------------------------------------
 
+IntType& IntType::add(int i )
+{
+    value += i;
+    return *this;
+}
```

Comment:
`value` is a pointer.  you need to use pointer semantics, just like you are doing in the `add` functions that take non-primitives
- [ ] revise